zerver/lib/actions.py

```diff
@@ -1367,6 +1348,27 @@ def do_change_user_delivery_email(user_profile: UserProfile, new_email: str) ->
         event_time=event_time,
     )
 
+    def send_delivery_email_change_events() -> None:
+        # We notify just the target user (and eventually org admins, only
+        # when email_address_visibility=EMAIL_ADDRESS_VISIBILITY_ADMINS)
+        # about their new delivery email, since that field is private.
+        payload = dict(user_id=user_profile.id, delivery_email=new_email)
+        event = dict(type="realm_user", op="update", person=payload)
+        send_event(user_profile.realm, event, [user_profile.id])
+
+        if user_profile.avatar_source == UserProfile.AVATAR_FROM_GRAVATAR:
+            # If the user is using Gravatar to manage their email address,
+            # their Gravatar just changed, and we need to notify other
+            # clients.
+            notify_avatar_url_change(user_profile)
+
+        if user_profile.email_address_is_realm_public():
+            # Additionally, if we're also changing the publicly visible
+            # email, we send a new_email event as well.
+            send_user_email_update_event(user_profile)
+
+    transaction.on_commit(lambda: send_delivery_email_change_events())
```

Comment:
Or `transaction.on_commit(send_delivery_email_change_events)`.

Reply:
Ahh, yeah, that'd be nicer.